### ISSUE TITLE
`custom-error-definition` is complaining about options as second parameter when it shouldn't

### ISSUE THREAD

**Issue by alfaproject** [sindresorhus/eslint-plugin-unicorn#3374]
As the title says `custom-error-definition` is being a bit too eager complaining about the options parameter location.

We have loads of custom errors in our codebase because we rely on the class instances rather than error messages or codes to check for equality, so there's errors like

```js
export class TimeoutError extends FetchError {
  constructor(cause) {
    super('The request timed out', { cause });
    this.name = 'TimeoutError';
  }
}
```

and other similar patterns where the message is always hard-coded because it's by design, we don't want to do things like `throw new MissingPaymentIdentifier('...', { cause });`

The message there can only be one, there's no need for a message input.
